src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.c

```diff
@@ -46,80 +77,1172 @@ const ActorInit En_Kakasi_InitVars = {
+            }
+        }
+    } else {
+        EnKakasi_SetupIdleUnderground(this);
+    }
+}
+
+void EnKakasi_SetAnimation(EnKakasi* this, s32 index) {
+    this->animeIndex = index;
+    this->animeFrameCount = SkelAnime_GetFrameCount(&kakasiAnimations[this->animeIndex]->common);
+    // 1: regular playback speed, 0: starting frame
+    SkelAnime_ChangeAnim(&this->skelanime, kakasiAnimations[this->animeIndex], 1.0f, 0.0f, this->animeFrameCount,
+                         D_8097206C[this->animeIndex], -4.0f);
+}
+
+/*
+ * set by init, called... nowhere? left over from OOT?
+ */
+void EnKakasi_8096F88C(GlobalContext* globalCtx, EnKakasi* this) {
+    func_8013A530(globalCtx, &this->actor, 0x7, &this->actor.focus.pos, &this->actor.shape.rot, 280.0f, 1800.0f, -1);
+}
+
+void EnKakasi_CheckAnimateSFX(EnKakasi* this) {
+    if (this->animeIndex == ENKAKASI_ANIME_SIDEWAYS_SHAKING || this->animeIndex == ENKAKASI_ANIME_ARMS_CROSSED_STILL) {
+        if (func_801378B8(&this->skelanime, 1.0f) != 0 || func_801378B8(&this->skelanime, 8.0f) != 0) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_KAKASHI_SWING);
+        }
+    }
+    if (this->animeIndex == ENKAKASI_ANIME_HOPPING_REGULAR || this->animeIndex == ENKAKASI_ANIME_SLOWROLL) {
+        if (func_801378B8(&this->skelanime, 4.0f) != 0 || func_801378B8(&this->skelanime, 8.0f) != 0) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_KAKASHI_SWING);
+        }
+        if (func_801378B8(&this->skelanime, 1.0f) != 0 || func_801378B8(&this->skelanime, 9.0f) != 0 ||
+            func_801378B8(&this->skelanime, 16.0f) != 0) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_IT_KAKASHI_JUMP);
+        }
+        if (func_801378B8(&this->skelanime, 18.0f) != 0) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_KAKASHI_ROLL);
+        }
+    }
+    if (this->animeIndex == ENKAKASI_ANIME_SPIN_REACH_OFFER || this->animeIndex == ENKAKASI_ANIME_TWIRL) {
+        if (func_801378B8(&this->skelanime, 1.0f) != 0) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_KAKASH_LONGI_ROLL);
+        }
+    }
+}
+
+/*
+ * moves the player's position relative to scarecrow during song teach, also each frame of dance the night away
+ */
+void EnKakasi_CheckPlayerPosition(EnKakasi* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (globalCtx->sceneNum == SCENE_8ITEMSHOP) {
+        player->actor.world.pos.x = -50.0f;
+        player->actor.world.pos.z = 155.0f;
+    } else if (globalCtx->sceneNum == SCENE_TENMON_DAI) {
+        player->actor.world.pos.x = 60.0f;
+        player->actor.world.pos.z = -190.0f;
+    }
+
+    Math_SmoothStepToS(&player->actor.shape.rot.y, (this->actor.yawTowardsPlayer + 0x8000), 5, 1000, 0);
+}
+
+/*
+ * this goes off every frame of dancing the night away, and song teaching
+ * something to do with cutscene camera?
+ */
+void func_8096FAAC(EnKakasi* this, GlobalContext* globalCtx) {
+    if (this->cutsceneCamId != 0) { // SUBCAM_FREE
```

Comment:
```suggestion
    if (this->cutsceneCamId != 0) {
```
We've dropped the `SUBCAM_FREE` macro in favor of just keeping it to `CAM_ID_MAIN`.